arch/x86/64/include/ArchMemory.h

```diff
@@ -4,143 +4,120 @@
+     * @param vaddress_to_check The virtual address to check
+     * @return True if mapping exists, false otherwise (accessing it would result in a pagefault)
+     */
+    pointer checkAddressValid(uint64 vaddress_to_check);
+
+    /**
+     * Fills out an ArchMemoryMapping object by translating the given
+     * virtual page and collecting infos from the page tables along the way.
+     * @param pml4 The pml4 of the ArchMemory object for which to translate the vpage
+     * @param vpage The virtual page to resolve
+     * @return Returns a completely or partially filled out ArchMemoryMapping object
+     */
+    static const ArchMemoryMapping resolveMapping(uint64 pml4, uint64 vpage);
+
+    const ArchMemoryMapping resolveMapping(uint64 vpage);
+
+    static uint64 get_PPN_Of_VPN_In_KernelMapping(uint64 virtual_page,
+                                                  uint64 *physical_page, uint64 *physical_pte_page=nullptr);
+
+    static void mapKernelPage(uint64 virtual_page, uint64 physical_page);
+
+    static void unmapKernelPage(uint64 virtual_page);
+
+    static PageMapLevel4Entry* getRootOfKernelPagingStructure();
+
+  private:
+    /**
+     * Adds a PML4Entry, PDPTEntry, PDEntry or PTEntry to the given PML4, PDPT, PD or PT respectively.
+     * (In other words, adds the reference to a new page table to a given page directory, for example.)
+     *
+     * @param map_ptr identity address of the physical page containing the target page table.
+     * @param index Index of the PML4Entry, PDPTEntry, PDEntry or PTEntry to be added.
+     * @param ppn physical page number of the new PDPT, PD, PT or page.
+     * @param bzero if true, the content of the newly inserted physical page is set to 0
+     * @param size if true, a 1G or 2M page is inserted, otherwise a PD or PT respectively.
+     * @param user_access if true, threads in user mode may access the inserted page. Otherwise, a kernel page is mapped
+     * @param writable if true, the inserted pages are writable
+     */
+    template <typename T> static void insert(pointer map_ptr, uint64 index, uint64 ppn, uint64 bzero, uint64 size, uint64 user_access, uint64 writeable);
+
+    /**
+     * Removes a PML4Entry, PDPTEntry, PDEntry or PTEntry from a given PML4, PDPT, PD or PT if it is present
+     * in the first place. This is done by setting the entry to present = 0 and clearing all other bits.
+     *
+     * @param map_ptr identity address of the physical page containing the target page table.
+     * @param index Index of the PML4Entry, PDPTEntry, PDEntry or PTEntry to be removed.
+     * @return True if the table map_ptr is full of zeroes and thus able to be freed.
+     */
+    template<typename T> static bool checkAndRemove(pointer map_ptr, uint64 index);
+
+    ArchMemory(ArchMemory const &src); /// Prevents accidental copying, may be moved to public
+    ArchMemory &operator=(ArchMemory const &src);
```

Comment:
It would be better to just make them public and mark these as explicitly deleted (with a comment that explains that they can be implemented if needed for e.g. fork)